### PR TITLE
Add ADPCM codec for Mac client

### PR DESCRIPTION
**Pull request by DavBfr**
need to build with -DWITH_MACAUDIO

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test this nice pr

**Comment by bmiklautz** [comment]
@freerdp-bot test

**Comment by hardening** [comment]
the @freerdp-bot only listen to the voice of its master ;)

File: channels/rdpsnd/client/mac/rdpsnd_mac.c [hunk 1]
```diff
@@ -205,6 +230,10 @@ static BOOL rdpsnd_mac_format_supported(rdpsndDevicePlugin* device, AUDIO_FORMAT
         {
                 return FALSE;
         }
+        else if (format->wFormatTag == WAVE_FORMAT_ADPCM || format->wFormatTag == WAVE_FORMAT_DVI_ADPCM)
```
Comment (hardening) [inline channels/rdpsnd/client/mac/rdpsnd_mac.c]: Not your code, but would you change all this to a nice switch() statement ?

**Comment by hardening** [comment]
LGTM

**Comment by akallabeth** [comment]
LGTM +1

If you add a ```cmake_dependent_option``` for ```WITH_MACAUDIO``` the pull can be merged.

**Comment by DavBfr** [comment]
I would rather add a
`set (WITH_MACAUDIO ON CACHE BOOL "Core audio"`
in ci/cmake-preloads/config-macosx.txt

**Comment by akallabeth** [comment]
@DavBfr The idea would be to automatically activate the flag if the build is for OS X or iOS (but not others). The ci scripts are more of a specialized configuration for our automated build system.

**Comment by DavBfr** [comment]
yes but there is not such thing for linux or windows.

**Comment by akallabeth** [comment]
@davbfr well, one platform a time ;)

**Comment by akallabeth** [comment]
@DavBfr Something like ```cmake_dependent_option(WITH_MACAUDIO "Enable OSX/iOS sound backend" ON "APPLE" OFF)``` in ```channels/ConfigOptions.cmake``` just like the one for windows.

**Pull request merged**